jgrapht-core/src/test/java/org/jgrapht/TestUtil.java

```diff
@@ -19,61 +19,104 @@
-        constructGraph(graph, edges);
-        return graph;
-    }
+  public static <V> Graph<V, DefaultEdge> createUndirected(V[][] edges) {
+    Graph<V, DefaultEdge> graph = new DefaultUndirectedWeightedGraph<>(DefaultEdge.class);
+    constructGraph(graph, edges);
+    return graph;
+  }
 
-    public static Graph<Integer, DefaultEdge> createUndirected(int[][] edges) {
-        Graph<Integer, DefaultEdge> graph = new DefaultUndirectedWeightedGraph<>(DefaultEdge.class);
-        constructGraph(graph, edges);
-        return graph;
-    }
+  public static Graph<Integer, DefaultEdge> createUndirected(int[][] edges) {
+    Graph<Integer, DefaultEdge> graph = new DefaultUndirectedWeightedGraph<>(DefaultEdge.class);
+    constructGraph(graph, edges);
+    return graph;
+  }
 
-    public static Graph<Integer, DefaultEdge> createDirected(int[][] edges) {
-        Graph<Integer, DefaultEdge> graph = new DefaultDirectedWeightedGraph<>(DefaultEdge.class);
-        constructGraph(graph, edges);
-        return graph;
-    }
+  public static Graph<Integer, DefaultEdge> createDirected(int[][] edges) {
+    Graph<Integer, DefaultEdge> graph = new DefaultDirectedWeightedGraph<>(DefaultEdge.class);
+    constructGraph(graph, edges);
+    return graph;
+  }
 
-    public static <V> Graph<V, DefaultEdge> createDirected(V[][] edges) {
-        Graph<V, DefaultEdge> graph = new DefaultDirectedWeightedGraph<>(DefaultEdge.class);
-        constructGraph(graph, edges);
-        return graph;
-    }
+  public static <V> Graph<V, DefaultEdge> createDirected(V[][] edges) {
+    Graph<V, DefaultEdge> graph = new DefaultDirectedWeightedGraph<>(DefaultEdge.class);
+    constructGraph(graph, edges);
+    return graph;
+  }
 
-    public static Graph<Integer, DefaultEdge> createPseudograph(int[][] edges){
-        Graph<Integer, DefaultEdge> graph = new WeightedPseudograph<>(DefaultEdge.class);
-        constructGraph(graph, edges);
-        return graph;
-    }
+  public static Graph<Integer, DefaultEdge> createPseudograph(int[][] edges) {
+    Graph<Integer, DefaultEdge> graph = new WeightedPseudograph<>(DefaultEdge.class);
+    constructGraph(graph, edges);
+    return graph;
+  }
 
-    public static <V> Graph<V, DefaultEdge> createPseudograph(V[][] edges){
-        Graph<V, DefaultEdge> graph = new WeightedPseudograph<>(DefaultEdge.class);
-        constructGraph(graph, edges);
-        return graph;
-    }
+  public static <V> Graph<V, DefaultEdge> createPseudograph(V[][] edges) {
+    Graph<V, DefaultEdge> graph = new WeightedPseudograph<>(DefaultEdge.class);
+    constructGraph(graph, edges);
+    return graph;
+  }
+
+  @SafeVarargs
+  public static <V, E> void addVertices(final Graph<V, E> graph, final V... vertices) {
+    for (V vertex : vertices)
+      graph.addVertex(vertex);
+  }
+
+  @SafeVarargs
+  public static <V, E> void addEdges(final Graph<V, E> graph, final V... vertices) {
```

Comment:
When I look at the test code that uses this, it's hard to actually read the edges in the big list of vertices.

Pair<V> ... vertexPairs would be OK, and then Pair.of(A, B) to instantiate.  It's more verbose, but significantly better for test readability.

Reply:
The goal for this helper method was to make creating graphs for tests as concise as possible. I tried the pair idea before but found it too wordy. As for readability, you got a point there. Conciseness should not exist at the expense of readability. Pairs are also more type-safe, but also incur creating more wrapper objects. I am considering changing it, let me think about it for a little while. 😀

Reply:
I implemented it according to your suggestion. Thanks for the helpful advice.